File: my_house_app/lib/models/agent.dart
class Agent {
  final int id;
  final String agentName;
  final String region;
  final String phone;
  final String status;

  Agent({
    required this.id,
    required this.agentName,
    required this.region,
    required this.phone,
    required this.status,
  });

  factory Agent.fromJson(Map<String, dynamic> json) {
    return Agent(
      id: json['id'] as int,
      agentName: json['agent_name'] ?? '',
      region: json['region'] ?? '',
      phone: json['phone'] ?? '',
      status: json['status'] ?? '',
    );
  }
}

File: my_house_app/backend/mkorishop/app/Models/Agent.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Agent extends Model
{
    use HasFactory;

    protected $fillable = [
        'id',
        'agent_name',
        'region',
        'phone',
        'status',
        'created_by',
    ];

    public function creator()
    {
        return $this->belongsTo(CustomUser::class, 'created_by');
    }
}
